File: Proj/texasHoldem_Ver1.4/Player.php
<?php

/*
 * File:   Player.php
 * Author: Byoung Mo Lee
 * Created on September 24, 2019 01:15 AM
 * Purpose: Texas Holdem
 * 
 */
include ("Card.php");

class Player {

    private $pName;
    private $nBalance;
    private $myCards;
    private $active=1;
    
    

    public function Player() {
        $this->setMyCards();
        //$this->pName="";
        $this->nBalance=1000;
    }

    public function setMyCards() {
        $this->myCards = array();
    }


    public function getMyCards() {
        return $this->myCards;
    }
    
    public function bet($b){
        $this->nBalance-=$b;
    }
    
    public function check($a){
        $this->nBalance-=$a;
    }
    
    public function fold(){
        $this->active=0;
    }

    
}

File: Proj/proj1_texasHoldem/C++/Proj2/test/Player.cpp
/* 
 * File:   Player.cpp
 * Author: Byoung Mo Lee
 * Created on May 27, 2019 10:12 AM
 * Purpose: Player class for Texas Holdem
 * 
 */

#include "Player.h"
#include "Hands.h"

int Player::getNumCards() const{
    int scan=0;
    bool found=0;
    
    while(!found && scan<7){
        if(myCards[scan].getFace()==-1) found=1;
        else scan++;
    }
    
    return scan;
}

void Player::updateMyHands(Player &obj){
    //delete[] obj.myFiveCards;
    //obj.myFiveCards=new Card [5];
    setHands(obj.myCards);
    //obj.hands=obj.getHands();
    //obj.myFiveCards=obj.getFiveCards();
    //obj.points=obj.getPoints();
}

int Player::getMyHands() const{
    return hands;
}
    
Card* Player::getMyFiveCards() const{
    return myFiveCards;
}

int Player::getMyPoints() const{
    return points;
}
void Player::addCards(Card *c, int n){ //mC=players.myCards, c=decked Cards, n=number of addition
    int pos=0;
    bool found=0;
    while(!found){
        if(myCards[pos].getFace()==-1){

            found=1;
        }
        else{
            pos++;
        }
    }
    //cout << myCards[pos].getFace() << endl; // for test
    //cout << pos << endl;                    // for test
    for(int i=0;i<n;i++){
        myCards[pos+i].setACard(c[i]);
        //c->displayCard();  //test for addCard function
    }
}

void Player::displayMyFiveCards(){
    while(myFiveCards->getFace()!=-1){
        myFiveCards->displayCard();
        myFiveCards++;
    }
}